crates/oxc_semantic/tests/integration/symbols.rs

```diff
@@ -355,3 +355,51 @@ fn test_ts_interface_heritage() {
     .has_number_of_references(1)
     .test();
 }
+
+#[test]
+fn test_arrow_implicit_return() {
+    SemanticTester::js("let i = 0; const x = () => i")
+        .has_root_symbol("i")
+        .has_number_of_reads(1)
+        .has_number_of_writes(0)
+        .test();
+
+    SemanticTester::js("let i = 0; const x = () => ++i")
+        .has_root_symbol("i")
+        .has_number_of_reads(1)
+        .has_number_of_writes(1)
+        .test();
+
+    SemanticTester::js("let i = 0; const x = () => { ++i }")
+        .has_root_symbol("i")
+        .has_number_of_reads(0)
+        .has_number_of_writes(1)
+        .test();
+
+    SemanticTester::js("let i = 0; const x = () => (0, ++i)")
+        .has_root_symbol("i")
+        .has_number_of_reads(1)
+        .has_number_of_writes(1)
+        .test();
+
+    SemanticTester::js("let i = 0; const x = () => (++i, 0)")
+        .has_root_symbol("i")
+        .has_number_of_reads(1)
+        .has_number_of_writes(1)
+        .test();
```

Comment:
Is this test correct? I don't think `i` is being read from here (but see my general comment - I'm not sure).

Reply:
TBH, I am not sure what `i` ReferenceFlags should be. The `x` function return result is 0 when it is called, so `i` should be `write` only.  I think we can try to make `i` only contain `write` if no tests are broken

Reply:
Let's try to get to the bottom of it! #5165

Reply:
IMO it's not, but there are other test cases that enforce this behavior. I mentioned this to you (@Dunqing) a while ago, when you made that change, and your rationale was that "we don't know what's a read and what's not a read, since things could be proxies and/or getters/setters"